src/main/java/org/folio/circulation/domain/Loan.java

```diff
@@ -220,37 +229,61 @@ public User getProxy() {
 
   Loan withProxy(User newProxy) {
     return new Loan(representation, item, user, newProxy, checkinServicePoint,
-      checkoutServicePoint, originalDueDate);
+      checkoutServicePoint, originalDueDate, loanPolicy);
   }
-  
+
   public Loan withCheckinServicePoint(ServicePoint newCheckinServicePoint) {
     return new Loan(representation, item, user, proxy, newCheckinServicePoint,
-      checkoutServicePoint, originalDueDate);
+      checkoutServicePoint, originalDueDate, loanPolicy);
   }
-  
+
   public Loan withCheckoutServicePoint(ServicePoint newCheckoutServicePoint) {
     return new Loan(representation, item, user, proxy, checkinServicePoint,
-      newCheckoutServicePoint, originalDueDate);
+      newCheckoutServicePoint, originalDueDate, loanPolicy);
   }
 
-  private void changeLoanPolicy(String newLoanPolicyId) {
+  public String getLoanPolicyId() {
+    return representation.getString("loanPolicyId");
+  }
+
+  public LoanPolicy getLoanPolicy() {
+    return loanPolicy;
+  }
+
+  public Loan withLoanPolicy(LoanPolicy newloanPolicy) {
+
+    return new Loan(representation, item, user, proxy, checkinServicePoint, checkoutServicePoint, originalDueDate, newloanPolicy);
+  }
+
+  private void setLoanPolicyId(String newLoanPolicyId) {
     if (newLoanPolicyId != null) {
       representation.put("loanPolicyId", newLoanPolicyId);
     }
   }
-  
+
+  private void setLoanPolicyName(String newLoanPolicyName) {
+
+    JsonObject loanPolicySummary = representation.containsKey(LoanProperties.LOAN_POLICY)
+            ? representation.getJsonObject(LoanProperties.LOAN_POLICY)
+            : new JsonObject();
+
+    loanPolicySummary.put("loanPolicyName", newLoanPolicyName);
+
+    representation.put(LoanProperties.LOAN_POLICY, loanPolicySummary);
```

Comment:
This representation is the internal representation of the loan, why is it being manipulated here rather than when mapping this record to a response (in the `LoanRepresentation` class)?

I acknowledge this is confusing. Keeping the JSON around within the domain was convenient, however it is creating more confusion than help at the moment :-/

Reply:
Agree, removed